packages/vite/src/plugins/plugin.ts

```diff
@@ -123,6 +123,16 @@ async function createNodesInternal(
     return {};
   }
 
+  const hasReactRouterConfig = siblingFiles.some((configFile) => {
+    const parts = configFile.split('.');
+    return (
+      parts[0] === 'react-router' && parts[1] === 'config' && parts.length > 2
+    );
+  });
+  if (hasReactRouterConfig) {
+    return {};
+  }
+
```

Comment:
This is too eager.

Users might still be using vitest to test their project which is inferred by the Vite plugin. 

This file shows how we avoid remix for the build and serve targets in the `buildViteTargets` function. 

We should replicate that to ensure vitest targets are still inferred correctly.

Reply:
Yeah, good point updated.